webapp/src/ts/services/db-sync.service.ts

```diff
@@ -317,17 +318,29 @@ export class DBSyncService {
     }
   }
 
+  private async migrateDb() {
+    try {
+      await this.migrationsService.runMigrations();
+    } catch (err) {
+      console.error('Error while running DB migrations', err);
+      this.sendUpdate({ state: SyncStatus.Unknown });
+      throw err;
```

Comment:
This may be a bit aggressive if someone is offline. Can we check the status and silently fail (or console.info) so they don't get spammed with errors and feedback docs just for being offline?

Reply:
I think that the effect is the same as if replication is attempted when the device is offline (except now replication will be blocked if it is). 
I definitely think that we should handle offline cases more gracefully, right now (without this migration code) the console log is riddled with "failed to fetch" errors.

Reply:
Fair point.